qiskit/extensions/quantum_initializer/squ.py

```diff
@@ -158,6 +164,11 @@ def validate_parameter(self, parameter):
             raise CircuitError(f"invalid param type {type(parameter)} in gate {self.name}")
 
 
+@deprecate_func(
+    since="0.45.0",
+    additional_msg="Instead, append a qiskit.circuit.library.UnitaryGate to the circuit.",
```

Comment:
Is it clear how to replace `squ` by `unitary`? or should some explanation be given?
Do both functions give the same output for 1-qubit unitaries?

Reply:
Replacing is straightforward, both allow the signature `Gate(unitary, qubit_index)`. SQU has two additional arguments that the unitary gate doesn't have:
* `mode` for the decomposition -- but there is only one option, so this argument is pointless 😄 (I assume the plan was to add other modes but that never happened)
* `up_to_diagonal`, which allows to implement the gate up to a diagonal. In the tests I did, however, this sometimes gave a decomposition with higher gate count than `UnitaryGate`, so I would think we can just remove it